eng/code-quality-reports/src/main/resources/checkstyle/checkstyle-suppressions.xml

```diff
@@ -333,17 +314,32 @@
   <suppress checks="[a-zA-Z0-9]*" files="[/\\]cosmos[/\\]rx[/\\]examples[/\\]"/>
   <suppress checks="[a-zA-Z0-9]*" files="[/\\]azure-cosmos-benchmark[/\\]"/>
   <suppress checks="com.azure.tools.checkstyle.checks.GoodLoggingCheck"
-            files="com.azure.cosmos.CosmosResponseDiagnostics"/>  
+            files="com.azure.cosmos.CosmosResponseDiagnostics"/>
   <suppress checks="com.azure.tools.checkstyle.checks.ThrowFromClientLoggerCheck"
             files=".*[/\\]cosmos[/\\]((?!(CosmosClient|CosmosScripts|CosmosAsyncDatabase|PartitionKeyDefinition
             |JsonSerializable|CosmosContainer|DatabaseAccount|CosmosContainerProperties|UniqueKeyPolicy|IncludedPath
-            |CosmosConflictProperties|CosmosDatabase|ConnectionPolicy|ThrottlingRetryOptions|ClientSideRequestStatistics)).*)"/> 
+            |CosmosConflictProperties|CosmosDatabase|ConnectionPolicy|ThrottlingRetryOptions|ClientSideRequestStatistics)).*)"/>
   <suppress checks="com.azure.tools.checkstyle.checks.(GoodLoggingCheck|ExternalDependencyExposedCheck)"
             files="com.azure.cosmos.JsonSerializable"/>  <!-- exposes logger from slf4j-->
   <suppress checks="javadoc" files="com.azure.cosmos.CompositePathSortOrder"/>
   <suppress checks="com.azure.tools.checkstyle.checks.EnforceFinalFieldsCheck" files="com.azure.cosmos.PartitionKey"/>
-   <suppress checks="com.azure.tools.checkstyle.checks.(EnforceFinalFieldsCheck|ExternalDependencyExposedCheck|NoImplInPublicAPI)" 
-            files="com.azure.cosmos.ClientSideRequestStatistics"/>   
-  <suppress checks="IllegalImport" 
+   <suppress checks="com.azure.tools.checkstyle.checks.(EnforceFinalFieldsCheck|ExternalDependencyExposedCheck|NoImplInPublicAPI)"
+            files="com.azure.cosmos.ClientSideRequestStatistics"/>
+  <suppress checks="IllegalImport"
             files="com.azure.cosmos.ClientSideRequestStatistics"/> <!-- Need OperatingSystemMXBean from sun to obtain cpu info -->
+
+  <!-- perf-test files are not public API, so they don't need to fulfill all requirements. -->
+  <suppress checks="Javadoc" files=".*[/\\]perf-test[/\\].*\.java"/>
+  <suppress checks="com.azure.tools.checkstyle.checks.ThrowFromClientLoggerCheck" files=".*[/\\]perf-test[/\\].*\.java"/>
+  <suppress checks="GoodLogging" files=".*[/\\]perf-test[/\\].*\.java"/>
```

Comment:
assuming, you meant perf-test to be the directory name, it should be perf-test-core

Reply:
This is for perf-tests that are not core, i.e. in storage. I'm adding another rule for perf.test.core now.